### ISSUE TITLE
What's up with all the pull requests, why aren't they merged already?

### ISSUE THREAD

**Issue by echonox** [arnesson/cordova-plugin-firebase#1060]
Some of the pull requests fixes some really bad issues, and some of them add some new cool features @robertarnesson why aren't they merged already?

**Comment by rodrigograca31** [comment]
Maybe he lost interest in this project. If that's the case please give someone else permissions to manage this project (Accept PRs, contribute code, etc)
@robertarnesson

**Comment by dpa99c** [comment]
@robertarnesson seems to be taking you an awful long time respond to PRs/issues for this plugin.

If you're no longer interested/able to maintain it, I would suggest you consider handing over the repo to the Cordova community where it can be better maintained since this is one of the most popular Cordova plugins. I'm sure the guys at Apache Cordova such as would be willing to take ownership (cc @janpio).

Worst case, we (the community) can fork this repo and publish it to npm under a new package ID e.g. `cordova-plugin-firebase-v2`. Then we can merge the PRs and fix the issues much faster.
Since I'm already a Cordova plugin dev and use this plugin in multiple projects, I will consider doing this if @robertarnesson continues to stay silent.

**Comment by rodrigograca31** [comment]
Yes this plugin is slowing down tuns of projects. It should be moved to the Cordova community.
@dpa99c he has an email in this profile: @arnesson we could email him....?
He is probably getting emails about this but not reading them ....

**Comment by dpa99c** [comment]
@rodrigograca31 yeah maybe he's muted Github so all these comments don't spam him 😆 
I'll try sending a (polite) email...

**Comment by janpio** [comment]
Apache Cordova usually does not take ownership of non core plugins (licencing and legal clearance would be a nightmare, as would the maintenance burden), but of course anyone from the community might do so themselves by either forking or finding an agreement with the original plugin owner @dpa99c . If there are multiple plugins that need ownership, maybe someone can also create a parent Github organization to "collect" those plugins.

**Comment by janpio** [comment]
That being said, please be nice to _any_ plugin author. Just because they published a free plugin you depend on, doesn't entitle you to anything. They are under no obligation to merge any pull requests, answer to any issues or do anything more than what they already did: Put their code on Github and let you use it. 

This dynamic of course changes when you pay someone to do some work, but all the details should then be covered by your legal contract. As I assume this is not the situation here, please consider your behavior.

**Comment by wilsolutions** [comment]
> @robertarnesson seems to be taking you an awful long time respond to PRs/issues for this plugin.
> 
> If you're no longer interested/able to maintain it, I would suggest you consider handing over the repo to the Cordova community where it can be better maintained since this is one of the most popular Cordova plugins. I'm sure the guys at Apache Cordova such as would be willing to take ownership (cc @janpio).
> 
> Worst case, we (the community) can fork this repo and publish it to npm under a new package ID e.g. `cordova-plugin-firebase-v2`. Then we can merge the PRs and fix the issues much faster.
> Since I'm already a Cordova plugin dev and use this plugin in multiple projects, I will consider doing this if @robertarnesson continues to stay silent.

You fixed the previous issue related to the April update, this ones got a bunch of people including myself, I would be happy to use your repo, wait...I'm already using it lol.

**Comment by tmk1991** [comment]
@dpa99c  - I really like that you're offering to grab this by the reigns. 

Theres also a firebase set of plugins maintained by @chemerisuk. Any shot of collaborating and maintaining only one firebase cordova plugin? Is there anyone that knows people from the Firebase team that would be willing to sponsor a plugin for everyone? 

I mentioned it over here: https://github.com/arnesson/cordova-plugin-firebase/issues/1033#issuecomment-491116010

**Comment by chemerisuk** [comment]
As people mentioned I do have set of Cordova plugins for firebase. Every plugin focuses on a particular SDK, so people include only dependencies they actually use. It's much easier to maintain such plugins too.

Completely agree with @janpio - respect all plugin authors and don't expect they respond to a new issue immediately. Even to this one :)

**Comment by echonox** [comment]
I would be happy to maintain the android part of the plugin as I don't have expertise in ios, and would be great to have more collaborators like @dpa99c and @chemerisuk

**Comment by echonox** [comment]
> nder no obligation to merge any pull requests, answer to any issues or do anything more than what they already did: Put their code on Github and let you use it.
> 
> This dynamic of course changes when you pay someone to do some work, but all the details should then be covered by your legal contract. As I assume this is not the situation here, please consider your behavior.

Indeed agree with you, but I don't think anyone here is disrespectful towards the author, in fact we are being too nice with this opened issue, there are 28 pull request as we speak the oldest is dating from 8 sept 2018 almost a year ago.. instead of just moving the code to a separate v2 repo and publishing it under different name we are asking to hand over the ownership or allow others to review and merge pulls, cause there are already bunch of firebase plugins out there, and instead of creating separate v2 we want to keep the popularity of this repo as it is. I my self did few plugins for the community and as a plugin owner yes I don't have any obligation to do the stuff you mentioned, but when building a something that's is being used by many apps, many of which depend on this plugin a lot.

It's because he did some much for this community and we also understand maintaining a repo can be hard, we ask him to either allow others to merge or we would have to move the project to a new repo as @dpa99c pointed.

**Comment by rodrigograca31** [comment]
Thanks @chemerisuk for reminding me that I probably dont need "all this crap" because I'm only using Firebase Analytics because Google is deprecating Analytics so I probably only need the analytics part of Firebase and can probably use some other plugin.
This plugin even made me add another one because of gradle problems.

That  said I'm actually using this plugin because it's the "recommended one" by Ionic v3 and v4:
https://ionicframework.com/docs/v3/native/firebase/

So, let's spend hours and hours changing plugin and solving new problems! yay! :tada: :cry: :sob: :joy: 

P.S. (IMHO The Ionic company could also step forward and help the community by maintaining some plugins themselves but they don't seem interested and having a company controlling stuff is ........ but that's just my opinion)

**Comment by robertarnesson** [comment]
Hey guys. I have not been maintaining this plugin for a long time. @soumak77 is the primary maintainer as of now. If anyone else is interested in helping out I will happily give access

**Comment by tmk1991** [comment]
I vote everyone collaborating on the ones @chemerisuk started. As mentioned, it’s more modular and light weight. That and he seems to have managed the whole version mis match issue with the different google service sdks. 

Just a thought.

**Comment by daviesdoclc** [comment]
What about this one? https://github.com/googleanalytics/firebase-analytics-cordova.  The guy IS from Google (although the code references chemerisuk).  We tried using the @chemerisuk one, but it started bring in lots of cocoa pods dependencies, etc.

**Comment by tmk1991** [comment]
@dpa99c & @chemerisuk  - Not sure if you guys have met up yet or not but maybe we can all hop on a hangout or something and talk through it? I'm not really familiar with making the plugins but as an end-user of them all I can bring up some of the pain points we face with our app and these plugins.

**Comment by wilsolutions** [comment]
I liked @tmk1991 suggestion, would be good to get together and plan, have some sort of roadmap or something like that. I'd like to help as well.

**Comment by amility** [comment]
Hi Guys, 
Is it still in discussion? Please update if there is any.

**Comment by hugoblanc** [comment]
This one seems great: [cordova-plugin-firebase-lib](https://github.com/wizpanda/cordova-plugin-firebase-lib)

Also, for Ionic users, I'm trying to integrate this fork into native [3057](https://github.com/ionic-team/ionic-native/pull/3057) don't understand why it is not merged yet
